shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -292,46 +292,27 @@ - (void)dealloc {
                                                       object:self
                                                     userInfo:nil];
 
-  // It will be destroyed and invalidate its weak pointers
-  // before any other members are destroyed.
-  _weakFactory.reset();
-
   /// nil out weak references.
   [_registrars
       enumerateKeysAndObjectsUsingBlock:^(id key, FlutterEngineRegistrar* registrar, BOOL* stop) {
         registrar.flutterEngine = nil;
       }];
 
-  [_labelPrefix release];
-  [_initialRoute release];
-  [_pluginPublications release];
-  [_registrars release];
   _binaryMessenger.parent = nil;
   _textureRegistry.parent = nil;
-  [_binaryMessenger release];
-  [_textureRegistry release];
-  _textureRegistry = nil;
-  [_isolateId release];
 
   NSNotificationCenter* center = [NSNotificationCenter defaultCenter];
   if (_flutterViewControllerWillDeallocObserver) {
     [center removeObserver:_flutterViewControllerWillDeallocObserver];
-    [_flutterViewControllerWillDeallocObserver release];
   }
   [center removeObserver:self];
```

Comment:
Optional: It's harmless to leave, but this line hasn't been necessary since we dropped iOS 8 support. (The thing above looks like it probably is still because of the way it was registered.)

Reply:
Will followup in another PR just to "minimise" the change in this patch, which admittedly there is nothing minimal about at the moment. Mostly just want to work through why it's unnecessary, and it's currently to early in the morning for my brain to do that.

Reply:
Totally fair :)

To save your brain work later, it's because of this:
> If your app targets iOS 9.0 and later or macOS 10.11 and later, and you used [addObserver:selector:name:object:](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver), you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it.

https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver